atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleQueryFactory.java

```diff
@@ -352,17 +353,17 @@ protected FullQuery getRowsColumnRangeSubQuery(
             long ts,
             BatchColumnRangeSelection columnRangeSelection) {
         String query = " /* GET_ROWS_COLUMN_RANGE (" + tableName + ") */ "
-                + "SELECT * FROM ( SELECT m.row_name, m.col_name, max(m.ts) as ts"
+                + "SELECT s.row_name, s.col_name, s.ts" + getValueSubselect("s", true)
+                + " FROM ( SELECT m.row_name, m.col_name, max(m.ts) as ts"
+                +         getValueSubselectForGroupBy("m")
                 + "   FROM " + tableName + " m"
                 + "  WHERE m.row_name = ?"
                 + "    AND m.ts < ? "
                 + (columnRangeSelection.getStartCol().length > 0 ? " AND m.col_name >= ?" : "")
                 + (columnRangeSelection.getEndCol().length > 0 ? " AND m.col_name < ?" : "")
                 + " GROUP BY m.row_name, m.col_name"
-                + " ORDER BY m.col_name ASC ) WHERE rownum <= " + columnRangeSelection.getBatchHint();
-        FullQuery fullQuery = new FullQuery(wrapQueryWithIncludeValue("GET_ROWS_COLUMN_RANGE", query, true))
-                .withArg(row)
-                .withArg(ts);
+                + " ORDER BY m.row_name ASC, m.col_name ASC ) s WHERE rownum <= " + columnRangeSelection.getBatchHint();
+        FullQuery fullQuery = new FullQuery(query).withArg(row).withArg(ts);
```

Comment:
Might be beyond scope of this PR, but I'd really like if we had unit tests that asserted output of generated SQL